plugins/cloud-azure/src/main/java/org/elasticsearch/cloud/azure/storage/AzureStorageServiceImpl.java

```diff
@@ -175,12 +175,14 @@ public OutputStream getOutputStream(String container, String blob) throws URISyn
 
         CloudBlobContainer blob_container = client.getContainerReference(container);
         if (blob_container.exists()) {
-            for (ListBlobItem blobItem : blob_container.listBlobs(keyPath + prefix)) {
+            for (ListBlobItem blobItem : blob_container.listBlobs(keyPath + (prefix == null ? "" : prefix))) {
                 URI uri = blobItem.getUri();
                 logger.trace("blob url [{}]", uri);
-                String blobpath = uri.getPath().substring(container.length() + 1);
-                BlobProperties properties = blob_container.getBlockBlobReference(blobpath).getProperties();
-                String name = blobpath.substring(keyPath.length() + 1);
+                String blobPath = uri.getPath().substring(1 + container.length() + 1);
```

Comment:
We now have +2 instead of +1. Is it intentional? If so, can you add a comment to understand why `1 + .. + 1` is needed?